From b2197a57c1261591e34d92a6ad7444ea97819db6 Mon Sep 17 00:00:00 2001
From: Carlo Landmeter <clandmeter@gmail.com>
Date: Tue, 1 Mar 2016 15:32:52 +0100
Subject: [PATCH 4/8] Ensure correct return value type

When compiling with musl libc the return type will be incorrect.
---
 include/sys/zfs_context.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/sys/zfs_context.h b/include/sys/zfs_context.h
index cc626fd..e68223e 100644
--- a/include/sys/zfs_context.h
+++ b/include/sys/zfs_context.h
@@ -637,7 +637,7 @@ extern void delay(clock_t ticks);
 #define	maxclsyspri	-20
 #define	defclsyspri	0
 
-#define	CPU_SEQID	(pthread_self() & (max_ncpus - 1))
+#define	CPU_SEQID	((uintptr_t)pthread_self() & (max_ncpus - 1))
 
 #define	kcred		NULL
 #define	CRED()		NULL
-- 
2.7.2

